core/src/com/unciv/logic/civilization/ReligionManager.kt

```diff
@@ -87,19 +87,98 @@ class ReligionManager {
+            (200 + 100 * greatProphetsEarned * (greatProphetsEarned + 1) / 2) *
+                    civInfo.gameInfo.gameParameters.gameSpeed.modifier
+            ).toInt()
 
-    fun canGenerateProphet(): Boolean {
-        if (religion == null || !religion!!.hasPantheon()) return false // First get a pantheon, then we'll talk about a real religion
+    private fun canGenerateProphet(): Boolean {
+        if (religion == null || religionState == ReligionState.None) return false // First get a pantheon, then we'll talk about a real religion
         if (storedFaith < faithForNextGreatProphet()) return false
         // In the base game, great prophets shouldn't generate anymore starting from the industrial era
         // This is difficult to implement in the current codebase, probably requires an additional variable in eras.json
-        // Also only if you either [have founded a religion] or [the max amount of religions (players/2 + 1) has not been reached].
-        // As this is yet to be implemented, this function does almost nothing
         return true
     }
+
+    private fun generateProphet() {
+        val prophetSpawnChange = (5f + storedFaith - faithForNextGreatProphet()) / 100f
+
+        if (Random(civInfo.gameInfo.turns).nextFloat() < prophetSpawnChange) {
+            val birthCities =
+                civInfo.cities.filter { it.religion.getMajorityReligion() == religion!!.name }
+            val birthCity: CityInfo =
+                if (birthCities.isEmpty()) {
+                    if (religionState == ReligionState.Pantheon) civInfo.getCapital()
+                    else civInfo.cities.first { it.id == religion!!.holyCityId }
+                } else {
+                    birthCities.random()
+                }
+            val prophet = civInfo.addUnit("Great Prophet", birthCity)
+            if (prophet == null) return
+            prophet.religion = religion!!.name
+            prophet.abilityUsedCount["Religion Spread"] = 0
+            storedFaith -= faithForNextGreatProphet()
+        }
+    }
+
+    fun mayUseGreatProphetAtAll(prophet: MapUnit): Boolean {
+        if (religion == null) return false // First found a pantheon
+        if (religion!!.isMajorReligion()) return false // Already created a major religion
+        if (prophet.abilityUsedCount["Religion Spread"] != 0) return false // Already used its power for other things
+        if (civInfo.gameInfo.civilizations.count {
+                it.religionManager.religion != null && it.religionManager.religion!!.isMajorReligion()
+            }
+                >= civInfo.gameInfo.civilizations.count { it.isMajorCiv() } / 2 + 1
+            ) return false // Too bad, too many religions have already been founded.
+
+        return true
+    }
+
+    fun mayUseGreatProphetNow(prophet: MapUnit): Boolean {
+        if (!mayUseGreatProphetAtAll(prophet)) return false
+        if (prophet.getTile().getCity() == null) return false
```

Comment:
Prophets don't need to be in the city center?

Reply:
Just tested it, apparently they do